assets/javascripts/discourse/lib/date-utilities.js.es6

```diff
@@ -361,10 +365,45 @@ function eventLabel(event, args = {}) {
         }
       }
     }
+
+    pastDue = moment() > start;
+
+    if (siteSettings.events_support_deadlines && event.deadline) {
+      deadline = true;
```

Comment:
There has to be a component in discourse that does this kind of display, for example for use in the countdown in a topic timer. Are you sure there isn't an existing helper function in `discourse/lib/formatter.js` that does this?

Reply:
Unfortunately duration.humanize used for topic timers removes the detail by approximation which is not what we want.

Similar issue with https://github.com/discourse/discourse/blob/4bfe78d1e66fab345dda56f791e59c749d0de3a6/app/assets/javascripts/discourse/app/lib/formatter.js#L59

However, I've cleaned up the formatting removing commas and units when value is 0, e.g. no hours here so removed:

<img width="671" alt="image" src="https://github.com/user-attachments/assets/b997fdb6-b3bd-43b7-94a2-05966e00de81">